update-forms-ver.go

```diff
@@ -32,24 +33,36 @@ func main() {
 		log.Fatalf("could not get latest forms info: %v", err)
 	}
 	log.Printf("Found %s.", latest)
-	if err := verifyURL(latest.ArchiveURL); err != nil {
-		log.Fatalf("could not verify archive url: %v", err)
+	filename := fmt.Sprintf("Standard_Forms_%s.zip", latest.Version)
+	if err := downloadURL(latest.ArchiveURL, filename); err != nil {
+		log.Fatalf("could not download archive url: %v", err)
 	}
+	patAPIArchiveURL := fmt.Sprintf("%s%s", PatFormsAPIPath, filename)
+	// TODO: replace log statement with actually replacing latest.ArchiveURL
+	log.Printf("New archive URL would be %s", patAPIArchiveURL)
 	json.NewEncoder(os.Stdout).Encode(latest)
 }
 
-func verifyURL(url string) error {
-	resp, err := client.Head(url)
+func downloadURL(url string, filename string) error {
+	resp, err := client.Get(url)
 	if err != nil {
 		return err
 	}
 	defer resp.Body.Close()
 	switch resp.StatusCode {
 	case http.StatusFound, http.StatusOK:
-		return nil
+		break
 	default:
 		return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
 	}
+
+	out, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	_, err = io.Copy(out, resp.Body)
```

Comment:
Could we throw in a zip.Reader here to verify the zip archive integrity? 🤓 Then do a f.Seek(0,0) before calling io.Copy 🙂 Just a suggestion to make this even more robust.

Reply:
Done! I suppose this could be susceptible to a [zip bomb](https://en.wikipedia.org/wiki/Zip_bomb), but it's unlikely that one would be published on the Winlink website.